solidity/contracts/isms/hook/CCIPIsm.sol

```diff
@@ -0,0 +1,150 @@
+    /// @param _sourceChainSelector The selector of the destination chain.
+    /// @param _sender The address of the sender.
+    modifier onlyAllowlisted(uint64 _sourceChainSelector, address _sender) {
+        if (!allowlistedSourceChains[_sourceChainSelector])
+            revert SourceChainNotAllowlisted(_sourceChainSelector);
+        if (!allowlistedSenders[_sender]) revert SenderNotAllowlisted(_sender);
+        _;
+    }
+
+    // ============ Internal functions ============
+
+    /// handle a received message
+    function _ccipReceive(
+        Client.Any2EVMMessage memory any2EvmMessage
+    )
+        internal
+        override
+        onlyAllowlisted(
+            any2EvmMessage.sourceChainSelector,
+            abi.decode(any2EvmMessage.sender, (address))
+        ) // Make sure source chain and sender are allowlisted
+    {
+        // bytes memory lastReceivedPayload = abi.decode(any2EvmMessage.data, (bytes)); // abi-decoding of the sent payload
+        bytes32 lastReceivedPayload = abi.decode(any2EvmMessage.data, (bytes32));  
+        verifyMessageId(lastReceivedPayload); // verifyMessageId(bytes32)
+
+        emit MessageReceived(
+            any2EvmMessage.messageId,
+            any2EvmMessage.sourceChainSelector, // fetch the source chain identifier (aka selector)
+            abi.decode(any2EvmMessage.sender, (address)), // abi-decoding of the sender address,
+            lastReceivedPayload
+        );
+    }
+
+    /**
+     * @notice Receive a message from the AbstractMessageIdAuthHook
+     * @dev Only callable by the authorized hook.
+     * @param messageId Hyperlane Id of the message.
+     */
+    function verifyMessageId(bytes32 messageId) public payable override {
+        require(
+            _isAuthorized(),
+            "AbstractMessageIdAuthorizedIsm: sender is not the hook"
+        );
+        require(
+            msg.value < 2 ** VERIFIED_MASK_INDEX,
+            "AbstractMessageIdAuthorizedIsm: msg.value must be less than 2^255"
+        );
+
+        verifiedMessages[messageId] = msg.value.setBit(VERIFIED_MASK_INDEX);
+        emit ReceivedMessage(messageId);
+    }
+
+
+    /**
+     * @notice Check if sender is authorized to message `verifyMessageId`.
+     */
+    function _isAuthorized() internal view override returns (bool) {
+        return msg.sender == TypeCasts.bytes32ToAddress(authorizedHook);
+    }
+
+    // ============ Public / External functions ============
+
+    /// @dev Updates the allowlist status of a source chain for transactions.
+    function addSourceChainToAllowlist(
+        uint64 _sourceChainSelector,
+        bool allowed
+    ) external onlyOwner {
+        allowlistedSourceChains[_sourceChainSelector] = allowed;
+    }
+
+    /// @dev Updates the allowlist status of a sender for transactions.
+    function addSenderToAllowlist(address _sender, bool allowed) external onlyOwner {
+        allowlistedSenders[_sender] = allowed;
```

Comment:
authorizedHook is only one and immutable. Like in the test if you set the hook as the authorizedHook and include it in allowlistedSenders, even if you want to update allowlistedSenders, authorizedHook cannot be changes making this function not very useful.

What you can do is:
- authorize using onlyAllowlisted and rm authorizedHook, making it incompatible with AbstractMessageIdAuthorizedIsm
- rm allowlistedSenders and use authorizedHook for onlyAllowlisted

Reply:
In case of CCIP, the sender is perceived the contract that has initiated the transaction on the source chain, so we have to explicitly add to allowlist all senders from all supported chains.